File: testing/javascript/tester.js
const JasmineTestCodeGenerator = require('./jasmine-test-code-generator'),
  JavascriptCodeJasmineTester = require('./javascript-code-jasmine-tester');

(async () => {
  let { SOLUTION_CODE: solutionCode, TEST_CODE: testCode, TEST_JSON: testJson } = process.env;
  let testResults;
  try {
    if (testJson !== undefined) {
      testCode = new JasmineTestCodeGenerator().generate(testJson);
    }
    testResults = await new JavascriptCodeJasmineTester().test(solutionCode, testCode);
  } catch (error) {
    process.exit(error instanceof SyntaxError ? 1 : 255);
  }
  process.stdout.write(testResults);
})();

File: testing/java/Tester.java
import org.json.JSONArray;

public class Tester {
  public static void main(String[] args) {
    String solutionCode = System.getenv("SOLUTION_CODE"), testCode = System.getenv("TEST_CODE");
    JSONArray testResults = null;
    try {
      testResults = new JavaCodeJUnitTester().test(solutionCode, testCode);
    } catch (Throwable throwable) {
      System.exit(
        throwable instanceof SyntaxException ? 1
        : throwable instanceof CompilationException ? 2
        : 255
      );
    }
    System.out.print(testResults);
  }
}
